src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/samplers/MetricsDBFileSampler.java

```diff
@@ -0,0 +1,83 @@
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.metrics.ReaderMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.stats.collectors.SampleAggregator;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.stats.emitters.ISampler;
+
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.FileSystems;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.Paths;
+import java.util.Objects;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class MetricsDBFileSampler implements ISampler {
+  private static final Logger LOG = LogManager.getLogger(MetricsDBFileSampler.class);
+  private final AppContext appContext;
+
+  public MetricsDBFileSampler(final AppContext appContext) {
+    Objects.requireNonNull(appContext);
+    this.appContext = appContext;
+  }
+
+  @Override
+  public void sample(SampleAggregator sampleCollector) {
+    String metricsdbPrefix = MetricsDB.getFilePrefix();
+
+    Path targetDirectoryPath = Paths.get(metricsdbPrefix).getParent();
+
+    String metricsdbPattern = "regex:" + metricsdbPrefix + "\\d+";
+    String metricsdbTarPattern = "regex:" + metricsdbPrefix.substring(0, metricsdbPrefix.length() - 1)
+        + ".tar.\\d+-\\d+-\\d+-\\d+-\\d+.gz";
+    PathMatcher metricsdbMatcher = FileSystems.getDefault().getPathMatcher(metricsdbPattern);
+    PathMatcher metricsdbTarMatcher = FileSystems.getDefault().getPathMatcher(metricsdbTarPattern);
+    DirectoryStream.Filter<Path> metricsdbFilter = metricsdbMatcher::matches;
+    DirectoryStream.Filter<Path> metricsdbTarFilter = metricsdbTarMatcher::matches;
+
+    int numUncompressedMetricsdbFiles = 0;
+    long sizeUncompressedMetricsdbFiles = 0;
+    int numMetricsdbFiles = 0;
+    long sizeMetricsdbFiles = 0;
+
+    try (DirectoryStream<Path> stream = Files.newDirectoryStream(targetDirectoryPath, metricsdbFilter)) {
+      for (Path entry : stream) {
+        try {
+          sizeUncompressedMetricsdbFiles += Files.size(entry);
+          numUncompressedMetricsdbFiles += 1;
+        } catch (IOException e) {
+          LOG.error("Failed to access metricsdb file - {}", entry);
+        }
+      }
+    } catch (IOException e) {
+      LOG.error("Failed to access metricsdb directory - {}", targetDirectoryPath);
+    }
+
+    try (DirectoryStream<Path> stream = Files.newDirectoryStream(targetDirectoryPath, metricsdbTarFilter)) {
+      for (Path entry: stream) {
+        try {
+          sizeMetricsdbFiles += Files.size(entry);
+          numMetricsdbFiles += 1;
+        } catch (IOException e) {
+          LOG.error("Failed to access metricsdb file - {}", entry);
+        }
+      }
+    } catch (IOException e) {
+      LOG.error("Failed to access metricsdb directory - {}", targetDirectoryPath);
+    }
```

Comment:
Can we avoid the nested `try-catch` blocks here? Let's move both the `DirectoryStream`  object creation to one 'try-catch' and then have inner 'try-catch' as now. Also, update the variable name from `stream` to `metricsdbStream` and `metricsdbTarStream` resp.

Reply:
Not a huge fan of removing the nested try-catch -- I don't want issues with one or two files corrupting the whole sampling. But yes, will make the other changes.

Reply:
I still dont understand why we have nested try-catch here.  Nested try-catch should be used in case the `catch` or `finally` block throws an exception. We have only Log statements in catch block here. 

There should be Unit Tests and Integrations Tests written to check for all failure scenarios, nested try-catch is hardly the way.

Reply:
Going to change this to use a blanket try-catch and simply return without sampling if any issues are found (say, if the stream contains a Path whose associated file is deleted in the middle of sampling).